src/net/unixsock_readmsg_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris
+
+package net
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func TestUnixConnReadMsgUnixSCMRightsCloseOnExec(t *testing.T) {
+	if !testableNetwork("unix") {
+		t.Skip("not unix system")
+	}
+
+	fds, err := syscall.Socketpair(syscall.AF_LOCAL, syscall.SOCK_STREAM, 0)
+	if err != nil {
+		t.Fatalf("Socketpair: %v", err)
+	}
+	writeFile := os.NewFile(uintptr(fds[0]), "parent-reads")
```

Comment:
You forgot to defer closing `writeFile`.
Also, you accidentally named `writeFile` as "parent-reads". That being said, the notion of "parent" here is a holdover from the original syscall unit test that involved a parent process and a child process, so perhaps these should just be named "write-socket" and "read-socket"?

Reply:
Got it. I have updated a fix for this, but I am still thinking how should I write the file of `os.Open(os.DevNull)` to `writeFile`

Reply:
done